### DISCUSSION TITLE
New Smart meter install.

### DISCUSSION THREAD

**Discussion by nivek1612** [BottlecapDave/HomeAssistant-OctopusEnergy#1089] [Q&A] [answered]
I am about to switch to smart meters. I know it can take a few weeks for data to start appearing though the WAN however if the octopus mini is installed at the same time as the meters does the data become available immediately or does that’s still take a few weeks 

Also I’m already using the integration to pull the rates (elec and gas) my dumb meter and then use HA to work out costs etc. After the smart meter install will I need to delete the api key and generate a new one  ?

**Comment by BottlecapDave** [comment, marked as answer]
> octopus mini is installed at the same time as the meters does the data become available immediately or does that’s still take a few week

The current consumption sensors will populate as soon as the Home Mini starts sending the data. However the previous consumption sensors won't populate until OE are able to get the data straight from your meter via your DNO.

> Also I’m already using the integration to pull the rates (elec and gas) my dumb meter and then use HA to work out costs etc. After the smart meter install will I need to delete the api key and generate a new one ?

You'll just need to reload the integration (or restart your HA instance) to see the new meters. This will create new devices and sensors (assuming your new meters have different serial numbers), so you'll need to manually remove the old versions and update any automations.

**Comment by nivek1612** [comment]
Super thanks

**Comment by nivek1612** [comment]
@BottlecapDave, I guess the only follow-up question is how quickly the new meters will get registered on my account.

**Reply by BottlecapDave** [reply]
That I don't know I'm afraid. I don't work for OE, so you'll need to ask them. Once it's available on the website, a reload of the integration should see the new meters.

**Comment by nivek1612** [comment]
The new meters and the mini where working before the installation engineer left

**Reply by BottlecapDave** [reply]
If you can see the meters on the website and they have an active tariff associated with it, then you should just need to reload the integration to see the new meters. If you didn't have a home mini before, then you'll need to go into the config of your account in the integration and tick the box to say you have a home mini.

**Reply by nivek1612** [reply]
All good thanks Dave. My reply was badly worded I meant the system is working and was from before the engineer left. Very quick set up

**Reply by BottlecapDave** [reply]
No problem and glad to hear you're all up and running.
